ee/clickhouse/models/property.py

```diff
@@ -158,9 +161,17 @@ def parse_prop_clauses(
                     f"{property_operator} 0 = 13"
                 )  # If cohort doesn't exist, nothing can match, unless an OR operator is used
             else:
-                person_id_query, cohort_filter_params = format_filter_query(cohort, idx)
-                params = {**params, **cohort_filter_params}
-                final.append(f"{property_operator} {table_name}distinct_id IN ({person_id_query})")
+
+                if person_properties_mode == PersonPropertiesMode.EXCLUDE:
```

Comment:
Q: Context on why cohorts needed to change this way?

Reply:
ClickhouseEventQuery was parsing props in a special way in `_get_props` and used an extra util function `_get_cohort_subquery` which was doing almost the same thing as parse_prop_clause handling of cohort filters. This wouldn't work since we're now relying on `parse_prop_grouped_clauses` for the entire filter parsing